core/state_transition.go

```diff
@@ -314,54 +311,63 @@ func (st *StateTransition) StakingTransitionDb() (usedGas uint64, err error) {
 		if err = rlp.DecodeBytes(msg.Data(), stkMsg); err != nil {
 			return 0, err
 		}
-		utils.Logger().Info().Msgf("[DEBUG STAKING] staking type: %s, txn: %+v", msg.Type(), stkMsg)
+		utils.Logger().Info().Msgf("[DEBUG STAKING] staking type: %s, gas: %d, txn: %+v", msg.Type(), gas, stkMsg)
 		if msg.From() != stkMsg.ValidatorAddress {
 			return 0, errInvalidSigner
 		}
+		if err := VerifyCreateValidatorStakingMsg(st.state, st.evm.EpochNumber, stkMsg); err != nil {
```

Comment:
Better put the Verify.... inside apply........, so the apply can never mistakenly commit an invalid message.

Reply:
sure, moved into apply....